apps/emqx_management/src/emqx_mgmt_data_backup.erl

```diff
@@ -0,0 +1,674 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_mgmt_data_backup).
+
+-export([
+    export/0,
+    export/1,
+    import/1,
+    import/2,
+    format_error/1
+]).
+
+-ifdef(TEST).
+-compile(export_all).
+-compile(nowarn_export_all).
+-endif.
+
+-elvis([{elvis_style, invalid_dynamic_call, disable}]).
+
+-include_lib("kernel/include/file.hrl").
+-include_lib("emqx/include/logger.hrl").
+
+-define(ROOT_BACKUP_DIR, "backup").
+-define(BACKUP_MNESIA_DIR, "mnesia").
+-define(TAR_SUFFIX, ".tar.gz").
+-define(META_FILENAME, "META.hocon").
+-define(CLUSTER_HOCON_FILENAME, "cluster.hocon").
+-define(CONF_KEYS, [
+    <<"mqtt">>,
+    <<"alarm">>,
+    <<"sysmon">>,
+    <<"sys_topics">>,
+    <<"limiter">>,
+    <<"log">>,
+    <<"persistent_session_store">>,
+    <<"crl_cache">>,
+    <<"conn_congestion">>,
+    <<"force_shutdown">>,
+    <<"flapping_detect">>,
+    <<"broker">>,
+    <<"force_gc">>,
+    <<"zones">>
+]).
+
+-define(DEFAULT_OPTS, #{is_cli => true}).
```

Comment:
Nit: it seems the flag is used for output, maybe inject the printer fun directly?

Reply:
Done, thanks